## Setup

In [ ]:
import pandas as pd

In [ ]:
ufos_path = 'YOUR PATH HERE'

In [ ]:
sightings = pd.read_csv(ufos_path)
sightings.head()

<ipython-input-3-e5c2fbc31669>:1: DtypeWarning: Columns (5,9) have mixed types. Specify dtype option on import or set low_memory=False.
  sightings = pd.read_csv(ufos_path)


datetime                  city state country     shape  \
0  10/10/1949 20:30            san marcos    tx      us  cylinder   
1  10/10/1949 21:00          lackland afb    tx     NaN     light   
2  10/10/1955 17:00  chester (uk/england)   NaN      gb    circle   
3  10/10/1956 21:00                  edna    tx      us    circle   
4  10/10/1960 20:00               kaneohe    hi      us     light   

  duration (seconds) duration (hours/min)  \
0               2700           45 minutes   
1               7200              1-2 hrs   
2                 20           20 seconds   
3                 20             1/2 hour   
4                900           15 minutes   

                                            comments date posted    latitude  \
0  This event took place in early fall around 194...   4/27/2004  29.8830556   
1  1949 Lackland AFB&#44 TX.  Lights racing acros...  12/16/2005    29.38421   
2  Green/Orange circular disc over Chester&#44 En...   1/21/2008        53.2   
3  My older brother and twin sister were leaving ...   1/17/2004  28.9783333   
4  AS a Marine 1st Lt. flying an FJ4B fighter/att...   1/22/2004  21.4180556   

    longitude  
0  -97.941111  
1  -98.581082  
2   -2.916667  
3  -96.645833  
4 -157.803611

## Compare the percentage of **all** sightings that occurred in the U.S. before 9/10/1993, versus on or after 9/10/1993:

In [ ]:
sightings['datetime'] = pd.to_datetime(sightings['datetime'],errors='coerce')

In [ ]:
before_x_files = sightings[sightings['datetime'] < '09/10/1993']
len(before_x_files[before_x_files['country'] == 'us']) / len(before_x_files)

0.7839315759955132

In [ ]:
on_after_x_files = sightings[sightings['datetime'] >= '09/10/1993']
len(on_after_x_files[on_after_x_files['country'] == 'us']) / len(on_after_x_files)

0.8142222712603094

**Before 9/10/1993, 78.4% of sightings in the dataset occurred in the U.S.**

**On and after 9/10/1993, 81.4% of sightings in the U.S. occurred in the dataset.**

## Compare the number of **overall** sightings in the 5 years prior to 9/10/1993 to the number of overall sightings in the 5 years on or after 9/10/1993:

In [ ]:
five_yrs_prior = sightings[sightings['datetime'].between('09/10/1988', '09/10/1993')]
len(five_yrs_prior)

1280

In [ ]:
five_yrs_on_after = sightings[sightings['datetime'].between('09/10/1993', '09/09/1998')]
len(five_yrs_on_after)

4627

**4,627 sightings occurred in the 5 years on or after 9/10/1993, versus just 1,280 in the 5 years before 9/10/1993. Perhaps the X Files was a truly global phenomenon?**

**The truth is out there!**